robot_smach_states/src/robot_smach_states/util/designators/arm.py

```diff
@@ -13,6 +15,24 @@
 __author__ = 'loy'
 
 
+def const_resolve(arm_designator, additional_properties):
+    """
+    Resolves the designator after adding properties. Note that the state is not altered.
+    :param arm_designator: ArmDesignator to which properties should be added.
+    :param additional_properties: Dict with the additional properties that are to be added.
+    :return: an arm with the desired properties and state
+    """
+
+    arm_designator_add_props = copy.copy(arm_designator)
+    for k, v in additional_properties.items():
+        if k in arm_designator_add_props.arm_properties:
+            for val in additional_properties[k]:
+                if val not in additional_properties[k]:
+                    arm_designator_add_props.arm_properties[k] += val
+        else:
+            arm_designator_add_props.arm_properties[k] = v
+    return arm_designator_add_props.resolve()
+
```

Comment:
Three things happen here and that is not clear from the name or docstring of this function.
1. You copy the designator (which, as stated above, interferes with locking behaviour)
2. You add requirements to your (copy of) arm designator
3. You resolve the copy and return it.
I would like to see a clear separation between these functions.

Also question. Where do you want to use this function?

Reply:
@LoyVanBeek I am not far enough that I will be merging #1051 into this branch I and would like your input on Peters comment. Do you have some time Tuesday to help me think about this?